test/Kestrel.FunctionalTests/Http2/PipeWriterStream.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    // Write only stream implementation for efficiently writing bytes from the request body
+    internal class PipeWriterStream : Stream
```

Comment:
You can probably just use RawStream with a null PipeReader directly in the tests instead of this.